### PR TITLE
Improve internal `cursorOffset` calculate

### PR DESCRIPTION
**Pull request by fisker**
Similar to #9012, this one is more tricky.

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/main/core.js [hunk 1]
```diff
@@ -171,68 +171,45 @@ function formatRange(text, opts) {
       ...opts,
       rangeStart: 0,
       rangeEnd: Infinity,
-      // track the cursor offset only if it's within our range
+      // Track the cursor offset only if it's within our range
       cursorOffset:
-        opts.cursorOffset >= rangeStart && opts.cursorOffset < rangeEnd
+        opts.cursorOffset > rangeStart && opts.cursorOffset < rangeEnd
```
Comment (fisker) [inline src/main/core.js:176]: Should not include `=`

File: src/main/core.js [hunk 2]
```diff
@@ -171,68 +171,45 @@ function formatRange(text, opts) {
...
-  const rangeLeft = text.slice(0, rangeStart);
-  const rangeRight = text.slice(rangeEnd);
 
   let { cursorOffset } = opts;
-  if (opts.cursorOffset >= rangeEnd) {
+  if (cursorOffset >= rangeEnd) {
     // handle the case where the cursor was past the end of the range
     cursorOffset =
-      opts.cursorOffset - rangeEnd + (rangeStart + rangeTrimmed.length);
+      opts.cursorOffset + (rangeTrimmed.length - rangeString.length);
```
Comment (fisker) [inline src/main/core.js:193]: Should be the same, but simpler

File: src/main/core.js (hunk 1 above)
Comment (evilebottnawi) [reply src/main/core.js:176]: Can we add test on this?

Comment (fisker) [reply src/main/core.js:176]: Sure, this is not breaking anything, just no need include `=`.

I'll  add one tomorrow.

Comment (fisker) [reply src/main/core.js:176]: Done https://github.com/prettier/prettier/pull/9013/commits/055870514fddb9f834e3695b6787501cdfa2f9cf

**Review by evilebottnawi — APPROVED** [review]

**Pull request merged**